### PR TITLE
visualizationResultToBytes converts anything that looks like a string

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Use `InteropLibrary.isString` and `asString` to convert any string value to `byte[]`

### Important Notes

Also contains a support for `Metadata.assertInCode` to help locating the right place in the code snippets.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
I tried various ways to write a test. Modify `RuntimeVisualizationsTest` to work with JavaScript object - it does, but it doesn't encode the value to `byte[]`. At the end I separated the conversion into own (Java) class and wrote a special `TruffleObject` that handles the _"to string"_ conversion.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala [hunk 1]
```diff
@@ -528,26 +522,20 @@ object ProgramExecutionSupport {
...
+    val bytes = VisualizationResult.visualizationResultToBytes(value)
+    if (bytes != null) {
+      Right(bytes)
+    } else {
+      Left(
+        new VisualisationException(
+          s"Cannot encode ${value.getClass} to byte array."
         )
+      )
     }
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]:
```suggestion
    Option(VisualizationResult.visualizationResultToBytes(value))
      .toRight(
        new VisualisationException(
          s"Cannot encode ${value.getClass} to byte array."
        )
      )
```

File: engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if (value instanceof String text) {
+      return text.getBytes(StandardCharsets.UTF_8);
+    }
+    if (value instanceof Text text) {
+      return visualizationResultToBytes(text.toString());
+    }
+    if (value instanceof WithWarnings warn) {
+      return visualizationResultToBytes(warn.getValue());
+    }
+    var iop = InteropLibrary.getUncached();
```
Comment (4e6) [inline engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java:26]: Not sure how `getUncached()` works. Is it worth making it a static member of the class?

**Review by 4e6 — APPROVED** [review]

File: build.sbt [hunk 3]
```diff
@@ -1612,6 +1612,7 @@ lazy val `runtime-with-polyglot` =
         "ENSO_TEST_DISABLE_IR_CACHE" -> "false"
       ),
       libraryDependencies ++= Seq(
+        "org.graalvm.sdk" % "graal-sdk" % graalVersion % "provided",
```
Comment (4e6) [inline build.sbt:1615]: I don't see where it is used in the project

File: engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java (hunk 2 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java:26]: This is the code:
```
    public static InteropLibrary getUncached() {
        return UNCACHED;
    }
```
e.g. it already returns static final field.

File: build.sbt (hunk 3 above)
Comment (JaroslavTulach) [reply build.sbt:1615]: `VerifyJavaScriptIsAvailableTest` is using the Graal SDK API. Without the dependency being _"provided"_ the test doesn't compile. _"provided"_ is probably the right scope - we don't need the dependency during runtime - it is provided by the GraalVM automatically when the test is running.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala]: OK, done in [7e44537](https://github.com/enso-org/enso/pull/4045/commits/7e445373b52864eac0bb2794bb5ed4541c7131b6)

**Review by jdunkerley — APPROVED** [review]

File: engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    if (iop.isString(value)) {
+      try {
+        return visualizationResultToBytes(iop.asString(value));
+      } catch (UnsupportedMessageException ex) {
+        // fallthru
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java:31]: Shouldn't this propagate via `IllegalStateException` instead?

**Review by hubertp — APPROVED** [review]

File: engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java (hunk 4 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java:31]: Having `asString` to throw an exception when `isString` returns `true` is an error in some other part of the system. Ignoring it doesn't cause any harm here. `VisualizationException` will likely be thrown anyway.

**Pull request merged**